### PR TITLE
sync pod status from leaf to root

### LINKED ISSUE #705: update  pod status from  leaf to root
**What would you like to be added**:

**Why is this needed**:

### PR DESCRIPTION
**Pull request by gao12312**
#### What type of PR is this?
/kind feature

#### What does this PR do?
update pod status from leaf to root

#### Which issue(s) does this PR fix?

Fixes #705
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/clustertree/cluster-manager/controllers/node_lease_controller.go [hunk 1]
```diff
@@ -89,6 +97,45 @@ func (c *NodeLeaseController) updateNodeStatus(ctx context.Context, n []*corev1.
...
+		}
+
+		rPodCopy := rootpod.DeepCopy()
+		if podutils.IsKosmosPod(rPodCopy) {
+			rPodCopy.Status = leafpod.Status
+			//klog.Errorf("leafpod Status.Conditions in root:%+v", leafpod.Status.Conditions)
+			//klog.Errorf("rPodCopy  status.Conditions after in root:%+v", rPodCopy.Status.Conditions)
+			podutils.FitObjectMeta(&rPodCopy.ObjectMeta)
+			rPodCopy.ResourceVersion = "0"
+			if err := c.root.Status().Update(ctx, rPodCopy); err != nil && !apierrors.IsNotFound(err) {
```
Comment (OrangeBao) [inline pkg/clustertree/cluster-manager/controllers/node_lease_controller.go]: use RetryOnConflict

Comment (gao12312) [reply pkg/clustertree/cluster-manager/controllers/node_lease_controller.go]: already add

File: pkg/clustertree/cluster-manager/controllers/pod/leaf_pod_controller.go [hunk 2]
```diff
@@ -32,36 +34,81 @@ type LeafPodReconciler struct {
...
+//				podCopy.ResourceVersion = "0"
+//				if err := r.RootClient.Status().Update(ctx, podCopy); err != nil && !apierrors.IsNotFound(err) {
+//					klog.V(4).Info(errors.Wrap(err, "error while updating pod status in kubernetes"))
+//					return reconcile.Result{RequeueAfter: utils.DefaultRequeueTime}, nil
+//				}
+//			}
+//		})
+//		return reconcile.Result{}, nil
+//	}
 func (r *LeafPodReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
```
Comment (OrangeBao) [inline pkg/clustertree/cluster-manager/controllers/pod/leaf_pod_controller.go:38]: Delete the annotated code above

File: pkg/clustertree/cluster-manager/cluster_controller.go [hunk 3]
```diff
@@ -60,6 +60,7 @@ type ClusterController struct {
 	ControllerManagers     map[string]manager.Manager
 	ManagerCancelFuncs     map[string]*context.CancelFunc
 	ControllerManagersLock sync.Mutex
+	SharedPodLock          sync.Mutex
```
Comment (OrangeBao) [inline pkg/clustertree/cluster-manager/cluster_controller.go]: This lock has some issues.

File: pkg/clustertree/cluster-manager/cluster_controller.go [hunk 4]
```diff
@@ -307,6 +310,7 @@ func (c *ClusterController) setupControllers(
 	leafPodController := podcontrollers.LeafPodReconciler{
 		RootClient: c.Root,
 		Namespace:  "",
+		PodLock:    &SharedPodLock,
 	}
```
Comment (duanmengkk) [inline pkg/clustertree/cluster-manager/cluster_controller.go]: The usage of the shared lock is incorrect.

Comment (gao12312) [reply pkg/clustertree/cluster-manager/cluster_controller.go]: already update

**Pull request closed without merging**